### ISSUE TITLE
sympy.sqrt can't multiply with literal int

### ISSUE THREAD

**Issue by nschloe** [microsoft/pyright#1230]
The following code yields and error when checked with pyright:
```
pyright --lib demo.py
```
```python
import sympy

sympy.sqrt(3) * 3
```
```
  3:1 - error: Operator "*" not supported for types "Pow" and "Literal[3]"
    Operator "*" not supported for types "Pow" and "Literal[3]" (reportGeneralTypeIssues)
```
It works fine with other sympy types, e.g., `Rational(2, 5)`.

The sympy folks suggest is a pyright issue (https://github.com/sympy/sympy/issues/20536). Feel free to close if not.

**Comment by erictraut** [comment]
Sympy doesn't appear to contain inlined types or packaged type stubs. You will need type information if you want to use it with a type checker.

**Comment by oscarbenjamin** [comment]
Maybe I don't understand how pyright works (just pinging over from the sympy issue).

I've used mypy and there the intention is that types are inferred on a best effort basis if there are no declarations. The type checker is not supposed to report an error because it couldn't infer the types: that's a false positive.

Of course there can be false positives. I made changes to eliminate a lot of those from the sympy codebase so that we could at least try out using mypy but it hasn't gone much further. I consider those false positives to be bugs in mypy though (a small number real bugs were also picked up).

In this case the error message from pyright:
```
  3:1 - error: Operator "*" not supported for types "Pow" and "Literal[3]"
```
is an incorrect statement because operator `*` is supported for Pow and 3.

**Comment by erictraut** [comment]
When mypy encounters an imported package that does not contain a "py.typed" file (i.e. the package doesn't claim to be typed), it treats all symbols imported from that package as "Any", which effectively disables any type checking associated with it. 

By default, pyright does the same. However, it has a mode that can be enabled by setting "useLibraryCodeForTypes" (or specifying "--lib" on the command line). If enabled, it will attempt to gather as much type information as possible from the code in the library. The result is type information that is sufficient in many cases for completion suggestions and function signature help, but it's often not good enough for type checking. 

Pylance, which is built on top of pyright, defaults the "useLibraryCodeForTypes" to true because most Pylance users are primarily interested in completion suggestions rather than type checking. If you're going to enable type checking (i.e. set "typeCheckingMode" to "basic") then I recommend setting "useLibraryCodeForTypes" to false (or avoiding the use of the "--lib" command-line option).

For best results, find type information for all of the packages you consume. Type stubs are available separately in some cases. In other cases, it may be feasible to create type stubs yourself. Or you can encourage the package maintainers to provide inline type information or packaged stubs in their next release.

**Comment by oscarbenjamin** [comment]
Okay, thanks for explaining. So pyright's `--lib` option should not be used with a library that does not have full typing annotations.

**Comment by erictraut** [comment]
You can try `--lib` with libraries that don't have full typing annotations, but you should expect that the inferred types will be relatively low quality. This can result in many false positive errors, like the one you reported here.
